modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/util/IgniteMath.java

```diff
@@ -270,6 +272,15 @@ public static byte divideExact(byte x, byte y) {
         return (byte) (x / y);
     }
 
+    /** Decimal division. */
+    public static BigDecimal divideDecimal(BigDecimal x, BigDecimal y, int scale) {
```

Comment:
let's mark types as nullable

Reply:
Removed this in favour of existing the `IgniteSqlFunctions decimalDivide`.